core-addon/IonCore.js

```diff
@@ -232,6 +238,57 @@ module.exports = class IonCore {
     }
   }
 
+  /**
+   * Handles messages coming in from the external website.
+   *
+   * @param {Object} message
+   *        The payload of the message.
+   * @param {runtime.MessageSender} sender
+   *        An object containing information about who sent
+   *        the message.
+   * @returns {Promise} The response to the received message.
+   *          It can be resolved with a value that is sent to the
+   *          `sender` or rejected in case of errors.
+   */
+  _handleWebMessage(message, sender) {
+    console.log(`IonCore - received web message ${message} from ${sender}`);
+
+    // TODO: check for https, check for host, etc. Do something more
+    // appropriate by properly creating a `URL` object.
```

Comment:
I think it's enough to create two URL objects and compare the `.origin`

Reply:
Oh, good suggestion. I updated this PR to cover it.